pkg/k8s/client.go

```diff
@@ -38,6 +38,11 @@ func CreateClients(config componentbaseconfig.ClientConnectionConfiguration) (cl
 			&clientcmd.ClientConfigLoadingRules{ExplicitPath: config.Kubeconfig},
 			&clientcmd.ConfigOverrides{}).ClientConfig()
 	}
+
+	if len(masterOverride) != 0 {
+		kubeConfig.Host = masterOverride
```

Comment:
Should we validate if the `masterOverride` is a valid URL or an IP address first?

Reply:
I see `clientset.NewForConfig(kubeConfig)` will do it and return an error if it's not a valid URL or an IP address, so we should be fine.